bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java

```diff
@@ -0,0 +1,796 @@
+                x -> GSON.fromJson(x, CommandConfirmation.class), listener);
+        sendCommand(request);
+    }
+
+    public void rewind(ResponseListener<CommandConfirmation> listener) {
+        String uri = "ssap://media.controls/rewind";
+        ServiceCommand<CommandConfirmation> request = new ServiceCommand<>(uri, null,
+                x -> GSON.fromJson(x, CommandConfirmation.class), listener);
+        sendCommand(request);
+    }
+
+    public void fastForward(ResponseListener<CommandConfirmation> listener) {
+        String uri = "ssap://media.controls/fastForward";
+        ServiceCommand<CommandConfirmation> request = new ServiceCommand<>(uri, null,
+                x -> GSON.fromJson(x, CommandConfirmation.class), listener);
+        sendCommand(request);
+    }
+
+    // APPS
+
+    public void getAppList(final ResponseListener<List<AppInfo>> listener) {
+        String uri = "ssap://com.webos.applicationManager/listApps";
+
+        ServiceCommand<List<AppInfo>> request = new ServiceCommand<>(uri, null,
+                jsonObj -> GSON.fromJson(jsonObj.get("apps"), new TypeToken<ArrayList<AppInfo>>() {
+                }.getType()), listener);
+
+        sendCommand(request);
+    }
+
+    public void launchAppWithInfo(AppInfo appInfo, ResponseListener<LaunchSession> listener) {
+        launchAppWithInfo(appInfo, null, listener);
+    }
+
+    public void launchAppWithInfo(final AppInfo appInfo, @Nullable JsonObject params,
+            final ResponseListener<LaunchSession> listener) {
+        String uri = "ssap://system.launcher/launch";
+        JsonObject payload = new JsonObject();
+
+        final String appId = appInfo.getId();
+
+        String contentId = null;
+
+        if (params != null) {
+            contentId = params.get("contentId").getAsString();
+        }
+
+        payload.addProperty("id", appId);
+
+        if (contentId != null) {
+            payload.addProperty("contentId", contentId);
+        }
+
+        if (params != null) {
+            payload.add("params", params);
+        }
+
+        ServiceCommand<LaunchSession> request = new ServiceCommand<>(uri, payload, obj -> {
+            LaunchSession launchSession = new LaunchSession();
+            launchSession.setService(this);
+            launchSession.setAppId(appId); // note that response uses id to mean appId
+            launchSession.setSessionId(obj.get("sessionId").getAsString());
+            launchSession.setSessionType(LaunchSessionType.App);
+            return launchSession;
+        }, listener);
+        sendCommand(request);
+    }
+
+    public void launchBrowser(String url, final ResponseListener<LaunchSession> listener) {
+        String uri = "ssap://system.launcher/open";
+        JsonObject payload = new JsonObject();
+        payload.addProperty("target", url);
+
+        ServiceCommand<LaunchSession> request = new ServiceCommand<>(uri, payload, obj -> new LaunchSession() {
+            {
```

Comment:
This generates a static inner class. Is that intended here?

Reply:
you are right. it should be done consistently as in launchAppWithInfo